docs/7-deployment/5-container.md

```diff
@@ -2,3 +2,609 @@
+      - CORS_ALLOWED_ORIGINS=https://app.example.com
+
+      # Authentication
+      - ALLOWED_ANONYMOUS=false
+      - SECRET_KEY=${SECRET_KEY}
+    volumes:
+      - ./workspace:/workspace
+    depends_on:
+      - postgres
+      - redis
+    restart: unless-stopped
+    healthcheck:
+      test: ["CMD", "wget", "--quiet", "--tries=1", "--spider", "http://localhost:15001/health"]
+      interval: 30s
+      timeout: 10s
+      retries: 3
+
+  postgres:
+    image: postgres:15-alpine
+    environment:
+      - POSTGRES_DB=hugr_core
+      - POSTGRES_USER=hugr
+      - POSTGRES_PASSWORD=hugr_password
+    volumes:
+      - postgres_data:/var/lib/postgresql/data
+    restart: unless-stopped
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready -U hugr"]
+      interval: 10s
+      timeout: 5s
+      retries: 5
+
+  redis:
+    image: redis:7-alpine
+    command: redis-server --requirepass redis_password --maxmemory 512mb --maxmemory-policy allkeys-lru
+    volumes:
+      - redis_data:/data
+    restart: unless-stopped
+    healthcheck:
+      test: ["CMD", "redis-cli", "--raw", "incr", "ping"]
+      interval: 10s
+      timeout: 5s
+      retries: 5
+
+volumes:
+  postgres_data:
+  redis_data:
+```
+
+## Building Custom Images
+
+If you need to customize the Docker images, you can build them from source. Clone the [hugr-lab/docker](https://github.com/hugr-lab/docker) repository:
+
+```bash
+git clone https://github.com/hugr-lab/docker.git
+cd docker
+```
+
+### Build Server Image
+
+```bash
+docker build -t ghcr.io/hugr-lab/server:latest -f server.dockerfile .
+```
+
+### Build Automigrate Image
+
+```bash
+docker build -t ghcr.io/hugr-lab/automigrate:latest -f automigrate.dockerfile .
+```
+
+### Build Management Image
+
+```bash
+docker build -t ghcr.io/hugr-lab/management:latest -f management.dockerfile .
+```
+
+### Using Docker Compose Build
+
+The repository includes a `docker-compose.yml` file for building:
+
+```bash
+docker-compose -f example.build.docker-compose.yml build
+```
+
+## Image Selection Guide
+
+Choose the right image based on your deployment scenario:
+
+| Deployment Scenario | Recommended Image | Notes |
+|---------------------|-------------------|-------|
+| Single-node production | `automigrate` | Automatic migrations on startup |
+| Development environment | `automigrate` | Simplifies schema updates |
+| Cluster work nodes | `server` | No automatic migrations needed |
+| Cluster management node | `management` | Manages cluster operations |
+
+## Kubernetes Deployment
+
+For Kubernetes deployments, see the [Cluster Deployment](./6-cluster.md) guide which includes Kubernetes manifests and configuration details.
+
+The [hugr-lab/docker](https://github.com/hugr-lab/docker) repository contains Kubernetes templates in the `k8s/cluster` directory.
+
+## Volume Mounts
+
+### Required Volumes
+
+- **`/data`** - Core database and local DuckDB files
+- **`/workspace`** - Schema definitions and data source catalogs
```

Comment:
It is not required to mount volumes, but to easily edit data sources schema definitions files from the host mount directory, also if you want to have access to the coredb duckdb file (if it duckdb) mount directory and setup correspond CORE_DB_PATH. To make secrets storage independent set up DB_HOME_DIRECTORY to persistence volume, to avoid erase it when the container is removed